### PR TITLE
Disable ODE DEBUG in release mode.

### PR DESCRIPTION
**Pull request by DavidMansolino**
**Description**
The ODE flag to disable debug mode is ``dNODEBUG`` and not ``NDEBUG``.

**Related Issues**
This pull-request fixes issue #2043

### CODE REVIEW

**Comment by DavidMansolino** [comment]
On my computer, I can measure with the performance logger and the PR2 world speed up of the step of ~3-3.5%.
Of course, this depends a lot on the world and computer used, but it at least proves the usefulness to add this no debug flag.

File: src/ode/ode/src/fluid_dynamics/ode_fluid_dynamics.cpp [hunk 1]
```diff
@@ -245,8 +245,9 @@ int dFluidDynamicsStep (dWorldID w)
       dMultiplyHelper1_331(rav, R, rav);
 
       //printf("linear speed %g  angular speed %g\n", rlvLength, avLength);
-
+#ifdef ODE_DEBUG
```
Comment (omichel) [inline src/ode/ode/src/fluid_dynamics/ode_fluid_dynamics.cpp:248]:
```suggestion
#ifndef dNODEBUG
```

File: src/ode/ode/src/ode_MT/util_MT.cpp [hunk 2]
```diff
@@ -243,6 +243,7 @@ void util_MT::checkBodyAndGeomConsistency(int clusterCount, ClusterNode** cluste
 
 void util_MT::checkClusterNodes(int clusterCount, dxClusterNode** clusterAABBs)
 {
+#ifdef ODE_DEBUG
```
Comment (omichel) [inline src/ode/ode/src/ode_MT/util_MT.cpp:246]:
```suggestion
#ifndef dNODEBUG
void util_MT::checkClusterNodes(int clusterCount, dxClusterNode** clusterAABBs)
{
```

File: src/ode/ode/src/ode_MT/util_MT.cpp [hunk 3]
```diff
@@ -253,4 +254,5 @@ void util_MT::checkClusterNodes(int clusterCount, dxClusterNode** clusterAABBs)
             prevNode = node;
         }
     }
+#endif
 }
```
Comment (omichel) [inline src/ode/ode/src/ode_MT/util_MT.cpp:258]:
```suggestion
}
#endif
```
And we should also add some #ifndef dNODEBUG in the `util_MT.h` file around this function declaration.

File: src/ode/ode/src/fluid_dynamics/ode_fluid_dynamics.cpp (hunk 1 above)
Comment (DavidMansolino) [reply src/ode/ode/src/fluid_dynamics/ode_fluid_dynamics.cpp:248]: (The PR is not yet ready for review, sorry bad old habit to switch to draft for triggering the tests)

**Comment by DavidMansolino** [comment]
I checked the macOS package created by Travis and there is no crash with the animated_skin anymore.

I also tried locally to find why an assertion was raised by uncommented the print of all the assertion-tested value:
https://github.com/cyberbotics/webots/blob/master/src/ode/ode/src/fluid_dynamics/ode_fluid_dynamics.cpp#L247-L317

But all the values looks normal from the development environment.

**Comment by omichel** [comment]
👀

**Review by omichel — APPROVED** [review]
It looks great. Thank you.

**Pull request merged**